File: obsPlugin/src/main/java/com/avid/central/obsplugin/InewsResponse.java
package com.avid.central.obsplugin;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.UUID;

/**
 * Created by Broadcast Media Solutions on 03/10/2015.
 */
@XmlRootElement(name = "inewsresponse")
@XmlAccessorType(XmlAccessType.NONE)
public class InewsResponse {
    @XmlElement
    private UUID id = null;
    @XmlElement(required = true)
    private String filename;
    @XmlElement(required = true)
    private String location;
    @XmlElement(required = true)
    private String rundown;
    @XmlElement(required = true)
    private String channelID;
    @XmlElement(required = true)
    private String title;
    @XmlElement(required = true)
    private String date;
    @XmlElement(required = true)
    private String day;
    @XmlElement(required = true)
    private String startTime;
    @XmlElement(required = true)
    private String endTime;
    @XmlElement(required = true)
    private String message;
    @XmlElement(required = true)
    private int result; // 0 = generic error, 1 = setup success, 2 = not authorized, 3 = configuration missing

    public UUID getID() {
        return this.id;
    }

    public void setID(UUID id) {
        this.id = id;
    }

    public String getFileName() {
        return this.filename;
    }

    public void setFileName(String filename) {
        this.filename = filename;
    }

    public String getLocation() {
        return this.location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getRundown() {
        return this.rundown;
    }

    public void setRundown(String rundown) {
        this.rundown = rundown;
    }

    public String getChannelID() {
        return this.channelID;
    }

    public void setChannelID(String channelID) {
        this.channelID = channelID;
    }

    public String getTitle() {
        return this.title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDate() {
        return this.date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getDay() {
        return this.day;
    }

    public void setDay(String day) {
        this.day = day;
    }

    public String getStartTime() {
        return this.startTime;
    }

    public void setStartTime(String startTime) {
        this.startTime = startTime;
    }

    public String getEndTime() {
        return this.endTime;
    }

    public void setEndTime(String endTime) {
        this.endTime = endTime;
    }

    public String getMessage() {
        return this.message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getResult() {
        return this.result;
    }

    public void setResult(int result) {
        this.result = result;
    }
}

File: obsPlugin/src/main/resources/static/js/export/datamodel/inewsResponse.js
(function (AV) {
    var InewsResponse;

    AV.ns("AV.obsPlugin.datamodel");

    InewsResponse = function (data) {
        this.id = null;
        this.filename = "";
        this.location = "";
        this.rundown = "";
        this.channelID = "";
        this.title = "";
        this.date = "";
        this.day = "";
        this.startTime = "";
        this.endTime = "";
        this.message = "";
        this.result = 0; // 0 = error, 1 = setup success, 2 = not authorized

        if (data) {
            if (data.id) {
                this.id = data.id;
            }
            this.filename = data.filename;
            this.location = data.location;
            this.rundown = data.rundown;
            this.channelID = data.channelID;
            this.title = data.title;
            this.date = data.date;
            this.day = data.day;
            this.startTime = data.startTime;
            this.endTime = data.endTime;
            this.message = data.message;
            this.result = data.result;
        }

        this.toHtml = function() {
            var html = new String("Exporting iNEWS rundown <b>");
            html = html.concat(this.rundown);
            html = html.concat("</b>OBSChannelID: <b>")
            html = html.concat(this.channelID);
            html = html.concat("</b><BR>Title: <b>")
            html = html.concat(this.title);
            html = html.concat("</b><BR>Date: <b>")
            html = html.concat(this.date);
            html = html.concat("</b><BR>Day: <b>")
            html = html.concat(this.day);
            html = html.concat("</b><BR>Start time: <b>")
            html = html.concat(this.startTime);
            html = html.concat("</b><BR>End time: <b>")
            html = html.concat(this.endTime);
            html = html.concat("</b><BR><BR>Warnings: <b>")
            html = html.concat(this.message == undefined ? "None" : this.message);
            html = html.concat("</b><BR><BR>Filename: <b>")
            html = html.concat(this.filename);
            html = html.concat("</b>");
            return html;
        }
    };


    function toJs()
    {
        return JSON.stringify(this);
    }

    AV.obsPlugin.datamodel.InewsResponse = InewsResponse;
}(AV));